rules/Php83/Rector/BooleanAnd/JsonValidateRector.php

```diff
@@ -0,0 +1,244 @@
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $funcCall = $this->matchJsonValidateArg($node);
+
+        if (! $funcCall instanceof FuncCall) {
+            return null;
+        }
+
+        if ($funcCall->isFirstClassCallable()) {
+            return null;
+        }
+
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return null;
+        }
+
+        $args = $funcCall->getArgs();
+        $positions = $this->argsAnalyzer->hasNamedArg($args)
+            ? $this->resolveNamedPositions($args)
+            : $this->resolveOriginalPositions($funcCall, $scope);
+
+        if ($positions === []) {
+            return null;
+        }
+
+        if (! $this->validateArgs($args, $positions)) {
+            return null;
+        }
+        $funcCall->name = new Name('json_validate');
+        $funcCall->args = $args;
+
+        return $funcCall;
+    }
+
+    public function providePolyfillPackage(): string
+    {
+        return PolyfillPackage::PHP_83;
+    }
+
+    public function matchJsonValidateArg(BooleanAnd $booleanAnd): ?FuncCall
+    {
+        // match: json_decode(...) !== null   OR   null !== json_decode(...)
+        if (! ($booleanAnd->left instanceof NotIdentical)) {
+            return null;
+        }
+
+        $decodeMatch = $this->binaryOpManipulator->matchFirstAndSecondConditionNode(
+            $booleanAnd->left,
+            fn ($node) => $node instanceof FuncCall && $this->isName($node->name, 'json_decode'),
+            fn ($node) => $node instanceof ConstFetch && $this->isName($node->name, 'null')
+        );
+
+        if (! $decodeMatch instanceof TwoNodeMatch) {
+            return null;
+        }
+
+        // match: json_last_error() === JSON_ERROR_NONE   OR   JSON_ERROR_NONE === json_last_error()
+        if (! ($booleanAnd->right instanceof Identical)) {
+            return null;
+        }
+
+        $errorMatch = $this->binaryOpManipulator->matchFirstAndSecondConditionNode(
+            $booleanAnd->right,
+            fn ($node) => $node instanceof FuncCall && $this->isName($node->name, 'json_last_error'),
+            fn ($node) => $node instanceof ConstFetch && $this->isName($node->name, 'JSON_ERROR_NONE')
+        );
+
+        if (! $errorMatch instanceof TwoNodeMatch) {
+            return null;
+        }
+
+        // always return the json_decode(...) call
+        $funcCall = $decodeMatch->getFirstExpr();
+        if (! $funcCall instanceof FuncCall) {
+            return null;
+        }
+        return $funcCall;
+    }
+
+    /**
+     * @param Arg[] $args
+     * @param int[]|string[] $positions
+     */
+    protected function validateArgs(array $args, array $positions): bool
+    {
+        foreach ($positions as $position) {
+            $arg = $args[$position] ?? '';
+            if ($arg instanceof Arg && $arg->name instanceof Identifier && $arg->name->toString() === 'flags') {
```

Comment:
Actually, you can get arg by name or position via `getArg($name, $position)`

https://github.com/nikic/PHP-Parser/blob/f7c23a43eee861070ab4e88819a4e76a611c7e4f/lib/PhpParser/Node/Expr/CallLike.php#L36-L59

Reply:
Thank you